### DISCUSSION TITLE
legend 不显示，不能触发 legendFilter 事件

### DISCUSSION THREAD

**Discussion by YY88Xu** [antvis/G2#5950] [Q&A]
### 问题描述

`import { Chart } from '@antv/g2';

const chart = new Chart({ container: 'container' });

chart
  .interval()
  .data([
    { genre: 'Sports', sold: 0 },
    { genre: 'Strategy', sold: 115 },
    { genre: 'Action', sold: 120 },
    { genre: 'Shooter', sold: 350 },
    { genre: 'Other', sold: 150 },
  ])
  .encode('x', 'genre')
  .encode('y', 'sold')
  .encode('color', 'genre')
  .style('minHeight', 50)
  .legend(false)

chart.render()

setTimeout(()=>{
  console.log('test')
  chart.emit('legend:filter', {
    data: { channel: 'color', values: ['Sports', 'Strategy'] },
  });
}, 2000)
`

### 重现步骤

chart
.legend(false)
希望也外部控制 legend:filter 事件触发

### 平台

- 操作系统: [macOS, Windows, Linux, React Native ...]
- 网页浏览器: [Google Chrome, Safari, Firefox]

**Comment by pearmini** [comment]
不没有 legend 不会去监听 `legend:filter` 事件，可以通过 `chart.changeData` 替代。
